File: cpp/AmmunitionShip.cpp
#include "AmmunitionShip.hpp"
#include <iostream>
AmmunitionShip::AmmunitionShip(Vector position, Vector direction, int length) : Ship(position, direction, length)
{
	std::cout << "Called ammunition ship constructor" << std::endl;
}
void AmmunitionShip::TakeDamage(const BulletType bulletType)
{
	if (bulletType == BulletType::Explosive)
	{
		hp = 0;
		std::cout << "Ammunition ship exploded" << std::endl;
		return;
	}
	hp--;
	if (!IsDestroyed())
		SpecialAbility();
	std::cout << "Ammunition ship damaged" << std::endl;
}
void AmmunitionShip::SpecialAbility()
{
	std::cout << "Ammunition using special ability" << std::endl;
	countOfSpecialAbility = 0;
}
AmmunitionShip& AmmunitionShip::operator=(Ship s)
{
	std::cout << "Assigned ship" << std::endl;
	AmmunitionShip ammunitionShip(s.GetPosition(), s.GetDirection(), s.GetLength());
	return ammunitionShip;
}

File: java/AmmunitionShip.java
public class AmmunitionShip extends Ship 
{
    public AmmunitionShip(Vector position, Vector direction, int length) 
    {
        super(position, direction, length);
        System.out.println("Called ammunition ship constructor");
    }    

    public void TakeDamage(BulletType bulletType)
    {
        if(bulletType == BulletType.Explosive)
        {
            hp = 0;
            System.out.println("Ammunition ship explosed");
            super.TakeDamage(bulletType);
            return;
        }
        hp--;
        if(!IsDestroyed())
            SpecialAbility();
        System.out.println("Ammunition ship damaged");
    }
    @Override
    public void SpecialAbility()
    {
        countOfSpecialAbility = 0;
        System.out.println("Ammunition ship using special ability");
    }
}
